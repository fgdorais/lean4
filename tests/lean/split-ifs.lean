/-! Test that the `split` tactic doesn't use `Classical.em` to split ifs.
-/

/- Basic example -/
theorem ittt1 (n : Nat) : (if n = 0 then 0 else 0) = 0 := by split <;> rfl

#print axioms ittt1

/- Example where the decidabilty instance has loose bound variables
  and split falls back on classical axioms (courtesy of Mario Carneiro). -/
theorem ittt2 (c : Prop) :
    (fun [Decidable c] => if c then 0 else 0) = (fun _ => 0) := by
  split <;> rfl

#print axioms ittt2
