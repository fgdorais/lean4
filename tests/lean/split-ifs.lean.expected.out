'ittt1' does not depend on any axioms
'ittt2' depends on axioms: [Classical.choice, Quot.sound, propext]
